force-app/tdtm/classes/TDTM_TriggerHandler.cls

```diff
@@ -377,11 +399,27 @@ public class TDTM_TriggerHandler {
 
         // now cache objects for future Update calls
         if (triggerAction == TDTM_Runnable.Action.AfterInsert || triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-            mapDescObjToMapOldForUpdates.put(describeObj, new Map<Id, SObject>(newList));
+            if (useDefaultCachingStrategry
+                || (useDefaultCachingStrategry == false && processedRecordsCount == newList.size())
+            ) {
+                mapDescObjToMapOldForUpdates.put(describeObj, new Map<Id, SObject>(newList));
+            
+            }
+
             //UTIL_Debug.debug(LoggingLevel.WARN, '**** cached newList: ' + newList);
         }
 
         return oldList;
     }
 
+    public static void enableOldListCachingEvaluation(TDTM_Runnable.Action triggerAction, Integer recordCount) {
+        if (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+            useDefaultCachingStrategry = false;
```

Comment:
I think this variable might need to be a Map<SObjectType, Boolean>, we don't want to disable the default caching strategy for all SObject Types, only RDs.